File: Algorithms/src/andy/javascript/level1/SportsUniform.js
// 프로그래머스 Level 1,체육복
function solution(n, lost, reserve) {
    var answer = 0;
    var arr = Array.from({length:n},() =>1);
	
    reserve.forEach(function(item) { arr[item-1]++ });
    lost.forEach(function(item) { arr[item-1]-- });
    arr.forEach(function(item,idx) { 
        if(item == 0) {
            if(arr[idx-1]>1) answer++;
            else if(arr[idx+1]>1) { answer++; arr[idx+1]--; }
        }
        else answer++;
    });

    return answer;
}

File: Algorithms/src/andy/level1/SportsUniform.java
package andy.level1;

import java.util.Arrays;

//프로그래머스 Level 1,체육복
public class SportsUniform {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] arr = new int[n];
        Arrays.fill(arr, 1);
        
        for(int d:reserve) arr[d-1]++;
        for(int d:lost) arr[d-1]--;
        for(int i=0;i<arr.length;i++) {
        	if(arr[i] == 0) {
        		if(i>0 && arr[i-1]>1) answer++;
        		else if(i<arr.length-1 && arr[i+1]>1) { 
        			answer++;
        			arr[i+1]--;
        		}
        	}
        	else 
        		answer++;
        }
        
        return answer;
    }
}
